dds/DCPS/DiscoveryBase.h

```diff
@@ -130,7 +132,7 @@ namespace OpenDDS {
     };
 
     template <typename DiscoveredParticipantData_>
-    class EndpointManager {
+    class EndpointManager : public virtual DCPS::RcEventHandler {
```

Comment:
Changing the base class of `EndpointManager` may require more changes than I see in this PR.  Making it an RcObject (via RcEventHandler) would impact how instances are created, use of RcHandle, etc.

Reply:
Is there a better (easier) way to implement a timer (PmfSporadicTask?) without changing the base class of EndpointManager?

Reply:
This is more of a design question, which is usually difficult to answer in this little text box in a PR review.  We can certainly discuss with the team outside of this review.

I haven't reviewed the related code yet, so I'm not sure what the "receiver" of this sporadic task needs to be -- maybe we could move it.

Also I'm not saying that changing the base class of EndpointManager is necessarily wrong, just that it has to be carried through to other uses of EndpointManager including derived classes.